src/builtin.c

```diff
@@ -58,24 +58,23 @@ BINOPS
 
 static jv type_error(jv bad, const char* msg) {
   char errbuf[15];
-  jv err = jv_invalid_with_msg(jv_string_fmt("%s (%s) %s",
-                                             jv_kind_name(jv_get_kind(bad)),
-                                             jv_dump_string_trunc(jv_copy(bad), errbuf, sizeof(errbuf)),
+  const char *badkind = jv_kind_name(jv_get_kind(bad));
+  jv err = jv_invalid_with_msg(jv_string_fmt("%s (%s) %s", badkind,
+                                             jv_dump_string_trunc(bad, errbuf, sizeof(errbuf)),
                                              msg));
-  jv_free(bad);
   return err;
 }
 
 static jv type_error2(jv bad1, jv bad2, const char* msg) {
   char errbuf1[15],errbuf2[15];
+  const char *badkind1 = jv_kind_name(jv_get_kind(bad1));
+  const char *badkind2 = jv_kind_name(jv_get_kind(bad2));
   jv err = jv_invalid_with_msg(jv_string_fmt("%s (%s) and %s (%s) %s",
-                                             jv_kind_name(jv_get_kind(bad1)),
-                                             jv_dump_string_trunc(jv_copy(bad1), errbuf1, sizeof(errbuf1)),
-                                             jv_kind_name(jv_get_kind(bad2)),
-                                             jv_dump_string_trunc(jv_copy(bad2), errbuf2, sizeof(errbuf2)),
```

Comment:
For understanding: was thinking first just remove the `jv_copy` and `jv_free` below would be enough but then i remembered arg order is undefined i think?

Reply:
Yep, it is undefined. Also, iirc most implementations evaluate right-to-left.